File: frontend/erpclink-web/src/app/features/patients/models/patient.models.ts
export enum Gender {
  Unknown = 0,
  Male = 1,
  Female = 2,
  Other = 3
}

export enum AllergySeverity {
  Unknown = 0,
  Mild = 1,
  Moderate = 2,
  Severe = 3,
  LifeThreatening = 4
}

export interface RegisterPatientRequest {
  firstName: string;
  middleName?: string | null;
  lastName: string;
  dateOfBirth: string;
  gender: Gender;
  nationalId?: string | null;
  phoneNumber: string;
  email?: string | null;
  addressLine1?: string | null;
  addressLine2?: string | null;
  city?: string | null;
  emergencyContactName?: string | null;
  emergencyContactPhone?: string | null;
  allowPossibleDuplicate?: boolean;
}

export interface UpdatePatientRequest {
  firstName: string;
  middleName?: string | null;
  lastName: string;
  dateOfBirth: string;
  gender: Gender;
  nationalId?: string | null;
  phoneNumber: string;
  email?: string | null;
  addressLine1?: string | null;
  addressLine2?: string | null;
  city?: string | null;
  emergencyContactName?: string | null;
  emergencyContactPhone?: string | null;
}

export interface SearchPatientsParams {
  query?: string;
  patientNumber?: string;
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
  nationalId?: string;
  isActive?: boolean;
  sortBy?: string;
  sortDescending?: boolean;
  page?: number;
  pageSize?: number;
}

export interface AddAllergyRequest {
  name: string;
  reaction?: string | null;
  severity: AllergySeverity;
  notes?: string | null;
}

export interface UpdateAllergyRequest {
  name: string;
  reaction?: string | null;
  severity: AllergySeverity;
  notes?: string | null;
}

export interface AllergyDto {
  id: string;
  name: string;
  reaction: string | null;
  severity: AllergySeverity;
  notes: string | null;
  isActive: boolean;
  createdAtUtc: string;
}

export interface PatientDto {
  id: string;
  organizationId: string;
  branchId: string;
  patientNumber: string;
  firstName: string;
  middleName: string | null;
  lastName: string;
  fullName: string;
  dateOfBirth: string;
  gender: Gender;
  nationalId: string | null;
  phoneNumber: string;
  email: string | null;
  addressLine1: string | null;
  addressLine2: string | null;
  city: string | null;
  emergencyContactName: string | null;
  emergencyContactPhone: string | null;
  isActive: boolean;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
  allergies: AllergyDto[];
  medicalHistory: MedicalHistoryItemDto[];
}

export interface MedicalHistoryItemDto {
  id: string;
  category: string;
  description: string;
  isActive: boolean;
  createdAtUtc: string;
}

export interface PatientListItemDto {
  id: string;
  patientNumber: string;
  fullName: string;
  dateOfBirth: string;
  phoneNumber: string;
  nationalId: string | null;
  isActive: boolean;
}

export interface RegisterPatientResult {
  patient: PatientDto;
  possibleDuplicates: PatientListItemDto[];
}

export const GENDER_LABELS: Record<Gender, string> = {
  [Gender.Unknown]: 'غير محدد',
  [Gender.Male]: 'ذكر',
  [Gender.Female]: 'أنثى',
  [Gender.Other]: 'أخرى'
};

export const ALLERGY_SEVERITY_LABELS: Record<AllergySeverity, string> = {
  [AllergySeverity.Unknown]: 'غير محدد',
  [AllergySeverity.Mild]: 'خفيف',
  [AllergySeverity.Moderate]: 'متوسط',
  [AllergySeverity.Severe]: 'شديد',
  [AllergySeverity.LifeThreatening]: 'مهدد للحياة'
};

File: src/Modules/Patients/ErpClink.Modules.Patients.Application/Patients/Models/PatientModels.cs
using ErpClink.BuildingBlocks.Application.Common;
using ErpClink.Modules.Patients.Domain.Patients;

namespace ErpClink.Modules.Patients.Application.Patients.Models;

public sealed record RegisterPatientRequest(
    string FirstName,
    string? MiddleName,
    string LastName,
    DateOnly DateOfBirth,
    Gender Gender,
    string? NationalId,
    string PhoneNumber,
    string? Email,
    string? AddressLine1,
    string? AddressLine2,
    string? City,
    string? EmergencyContactName,
    string? EmergencyContactPhone,
    bool AllowPossibleDuplicate = false);

public sealed record UpdatePatientRequest(
    string FirstName,
    string? MiddleName,
    string LastName,
    DateOnly DateOfBirth,
    Gender Gender,
    string? NationalId,
    string PhoneNumber,
    string? Email,
    string? AddressLine1,
    string? AddressLine2,
    string? City,
    string? EmergencyContactName,
    string? EmergencyContactPhone);

public sealed record SearchPatientsRequest(
    string? Query,
    string? PatientNumber,
    string? FirstName,
    string? LastName,
    string? PhoneNumber,
    string? NationalId,
    bool? IsActive,
    string? SortBy,
    bool SortDescending = false,
    int Page = 1,
    int PageSize = 20);

public sealed record AddAllergyRequest(
    string Name,
    string? Reaction,
    AllergySeverity Severity,
    string? Notes);

public sealed record UpdateAllergyRequest(
    string Name,
    string? Reaction,
    AllergySeverity Severity,
    string? Notes);

public sealed record AllergyDto(
    Guid Id,
    string Name,
    string? Reaction,
    AllergySeverity Severity,
    string? Notes,
    bool IsActive,
    DateTime CreatedAtUtc);

public sealed record MedicalHistoryItemDto(
    Guid Id,
    string Category,
    string Description,
    bool IsActive,
    DateTime CreatedAtUtc);

public sealed record PatientDto(
    Guid Id,
    Guid OrganizationId,
    Guid BranchId,
    string PatientNumber,
    string FirstName,
    string? MiddleName,
    string LastName,
    string FullName,
    DateOnly DateOfBirth,
    Gender Gender,
    string? NationalId,
    string PhoneNumber,
    string? Email,
    string? AddressLine1,
    string? AddressLine2,
    string? City,
    string? EmergencyContactName,
    string? EmergencyContactPhone,
    bool IsActive,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy,
    IReadOnlyList<AllergyDto> Allergies,
    IReadOnlyList<MedicalHistoryItemDto> MedicalHistory);

public sealed record PatientListItemDto(
    Guid Id,
    string PatientNumber,
    string FullName,
    DateOnly DateOfBirth,
    string PhoneNumber,
    string? NationalId,
    bool IsActive);

public sealed record RegisterPatientResult(
    PatientDto Patient,
    IReadOnlyList<PatientListItemDto> PossibleDuplicates);

public sealed record PagedPatientsResult(
    IReadOnlyList<PatientListItemDto> Items,
    int Page,
    int PageSize,
    int TotalCount)
{
    public static PagedPatientsResult From(PagedResult<PatientListItemDto> page) =>
        new(page.Items, page.Page, page.PageSize, page.TotalCount);
}
